edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/Gradebook.java

```diff
@@ -70,314 +156,35 @@ public Gradebook() {
-	 */
-	public Long getId() {
-		return id;
-	}
-	/**
-	 * @param id The id to set.
-	 */
-	public void setId(Long id) {
-		this.id = id;
-	}
-	/**
-	 * @return Returns the uid.
-	 */
-	public String getUid() {
-		return uid;
-	}
-	/**
-	 * @param uid The uid to set.
-	 */
-	public void setUid(String uid) {
-		this.uid = uid;
-	}
-	/**
-	 * @return Returns the name.
-	 */
-	public String getName() {
-		return name;
-	}
-	/**
-	 * @param name The name to set.
-	 */
-	public void setName(String name) {
-		this.name = name;
-	}
-	/**
-	 * @return Returns the locked.
-	 */
-	public boolean isLocked() {
-		return locked;
-	}
-	/**
-	 * @param locked The locked to set.
-	 */
-	public void setLocked(boolean locked) {
-		this.locked = locked;
-	}
-	/**
-	 * @return Returns the selectedGradeMapping.
-	 */
-	public GradeMapping getSelectedGradeMapping() {
-		return selectedGradeMapping;
-	}
-	/**
-	 * @param selectedGradeMapping The selectedGradeMapping to set.
-	 */
-	public void setSelectedGradeMapping(GradeMapping selectedGradeMapping) {
-		this.selectedGradeMapping = selectedGradeMapping;
-	}
-	/**
-	 * @return Returns the version.
-	 */
-	public int getVersion() {
-		return version;
-	}
-	/**
-	 * @param version The version to set.
-	 */
-	public void setVersion(int version) {
-		this.version = version;
-	}
-	
-	/**
-	 * Is the course grade to be shown at all?
-	 * @return boolean
-	 */
-    public boolean isCourseGradeDisplayed() {
-        return courseGradeDisplayed;
-    }
-   
-    public void setCourseGradeDisplayed(boolean courseGradeDisplayed) {
-        this.courseGradeDisplayed = courseGradeDisplayed;
-    }
-
-    public boolean isTotalPointsDisplayed() {
-      return totalPointsDisplayed;
-    }
-
-    public void setTotalPointsDisplayed(boolean totalPointsDisplayed) {
-      this.totalPointsDisplayed = totalPointsDisplayed;
-    }
-
-    /**
-	 * If the course grade is displayed, should the total points be displayed?
-	 * @return true/false if total points should be displayed
-	 */
-    public boolean isCoursePointsDisplayed() {
-		return coursePointsDisplayed;
-	}
-
-	public void setCoursePointsDisplayed(boolean coursePointsDisplayed) {
-		this.coursePointsDisplayed = coursePointsDisplayed;
-	}
-
-	/**
-	 * If the course grade is displayed, should the percentage be displayed?
-	 * @return true/false if percentage should be displayed
-	 */
-	public boolean isCourseAverageDisplayed() {
-      return courseAverageDisplayed;
-    }
-
-    public void setCourseAverageDisplayed(boolean courseAverageDisplayed) {
-      this.courseAverageDisplayed = courseAverageDisplayed;
-    }
 
     @Override
 	public String toString() {
         return new ToStringBuilder(this).
-        append("id", id).
-        append("uid", uid).
-        append("name", name).toString();
+        append("id", this.id).
+        append("uid", this.uid).
+        append("name", this.name).toString();
     }
 
     @Override
-	public boolean equals(Object other) {
```

Comment:
Lombok as @EqualsAndHashcode to make this simpler, you can control which fields should be used in the comparison.
For example to add an equals and hashcode on just the uid you could use:
`@EqualsAndHashcode(of = {"uid"})`

Reply:
Done. FYI `of` is being deprecated so new pattern is:
````
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@ToString(onlyExplicitlyIncluded = true)
public class SomeClass {
...
```
Then on the field:
````
@ToString.Include
@EqualsAndHashCode.Include
private String field;
````
👍